File: services/service2.go
package services

type Service2 struct {
	services *Services
}

func (s *Service2) DoSomething() int {
	// fmt.Println("Doing something from Service2")
	return 1
}

func (s *Service2) Query() {
	// fmt.Println("Querying from Service2")
}

File: src/services/service2.rs
use super::ServicesInner;

pub struct Service2 {}

impl Service2 {
    pub fn new() -> Service2 {
        Service2 {}
    }
}

impl ServicesInner<'_, &Service2> {
    pub fn do_something(&self) {
        println!("Service2::do_something()");
    }

    pub fn call_service_1(&self) {
        self.services.service1.do_something(42);
    }
}
